gcfgenv.go

```diff
@@ -212,7 +220,11 @@ func setGcfgWithEnvMap(ref reflect.Value, prefix string, env map[string]string)
 					if err != nil {
 						return err
 					}
-					f.Elem().Field(j).Set(newRef)
+					if f.Elem().Field(j).Kind() == reflect.Slice {
+						f.Elem().Field(j).Set(reflect.AppendSlice(f.Elem().Field(j).Elem(), newRef))
```

Comment:
One question I have here: this **appends** the environment variable to the existing slice -- should we instead append the existing slice to the environment variable? That would be more consistent with the general convention of the environment variables taking precedence over the configuration file.

Reply:
That makes sense. I feel like it is surprising behavior though. Since you're reading it into an existing array, overwriting / prepending against an existing list feels very counterintuitive IMO. 

I.e. to imagine you have `{"one": ["a"]}`
And then after env vars: `{"one": ["b", "c", "a"]}` just feels very unnatural to me 😅 

The precedence to me makes sense as "last in wins." But "last in... ends up prepending in front of the first in" does not logically follow IMO. It feels much more natural to say "last in wins... or last in appends to the end of whatever was there before"